libraries/collide2/CMakeLists.txt

```diff
@@ -0,0 +1,96 @@
+# Copyright (C) 2025 Benjamen Meyer and other Vega Strike contributors.
+#
+# https://github.com/vegastrike/Vega-Strike-Engine-Source
+#
+# This file is part of Vega Strike.
+#
+# Vega Strike is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Vega Strike is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Vega Strike. If not, see <https://www.gnu.org/licenses/>.
+
+MESSAGE("PROCESSING DIRECTORY: ${CMAKE_CURRENT_SOURCE_DIR}")
+
+LIST(APPEND
+    OPCOLLIDE_INCLUDES
+    # collide2 headers
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    # VS engine headers
+    #${CMAKE_SOURCE_DIR}/engine
```

Comment:
Please open an issue for refactoring VS out of collide library. It's no good separating it out if it refers to VS all over.

Reply:
1. collide2 is generally under a difference license (public domain from what I can tell... or rather the UNLICENSE) so at a minimum we gain that clarity.
2. agreed - there's a few things (gfx/quaternion.h) that seem to have gotten spread out. it'd be nice to make it more independent that way.

Filed #1042

Reply:
Looking at the build failures (https://github.com/vegastrike/Vega-Strike-Engine-Source/pull/1041#issuecomment-2687035496) I'll have to address some of this somehow... but my goal is to be minimal right now. Not sure if I'll fix up #1041 immediately or just move some checks around in CMake for the moment so Mac/Win builds succeed - more likely just move stuff around b/c it'll probably make more sense long term with other libraries being split out any way.